### PR TITLE
Add compat macros for PG15 signature changes

### PR DESCRIPTION
**Pull request by svenklemm**
PG15 refactors INSERT/UPDATE/DELETE code and changes the signatures
of some of the functions.

https://github.com/postgres/postgres/commit/25e777cf

### CODE REVIEW

**Comment by svenklemm** [comment]
With this applied timescaledb will compile against PG15

File: src/compat/compat.h [hunk 1]
```diff
@@ -624,7 +624,92 @@ pg_strtoint64(const char *str)
...
+#define ExecARUpdateTriggersCompat(estate,                                                         \
+								   resultRelInfo,                                                  \
+								   src_partinfo,                                                   \
+								   dst_partinfo,                                                   \
+								   tupleid,                                                        \
+								   oldtuple,                                                       \
+								   inewslot,                                                       \
+								   recheckIndexes,                                                 \
+								   transtition_capture,                                            \
+								   is_crosspart_update)                                            \
```
Comment (fabriziomello) [inline src/compat/compat.h:659]: In our code we always set this parameter as `false` so we can simplify this signature removing the `is_crosspart_update` and set `false` for prior pg15 version.

File: src/compat/compat.h [hunk 2]
```diff
@@ -624,7 +624,92 @@ pg_strtoint64(const char *str)
...
+#endif
+
+#if PG15_LT
+#define ExecBRUpdateTriggersCompat(estate,                                                         \
+								   epqstate,                                                       \
+								   resultRelInfo,                                                  \
+								   tupleid,                                                        \
+								   oldtuple,                                                       \
+								   slot,                                                           \
+								   tmfdp)                                                          \
```
Comment (fabriziomello) [inline src/compat/compat.h:679]: Same here for `tmfdp` that for prior pg15 version we pass `NULL`

**Review by fabriziomello — APPROVED** [review]

File: src/compat/compat.h (hunk 1 above)
Comment (svenklemm) [reply src/compat/compat.h:659]: With these compat macros we usually adopt the signature of the latest version to create the least friction moving forward and also to make it easy to replace the macro once we no longer need to support the older versions that required the macro.

**Review by konskov — APPROVED** [review]

**Pull request merged**